authLdap.php

```diff
@@ -356,6 +363,67 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
 }
 
 /**
+ * Get user's user id
+ *
+ * Returns null if username not found
+ *
+ * @param string $username username
+ * @param string user id, null if not found
+ */
+function authLdap_get_uid($username) {
+    global $wpdb;
+    static $cache = array();
+
+    if (!isset($cache[$username])) {
```

Comment:
It's also used in authLdap_user_role, so it's used twice. Could probably pass the uid into user_role instead of username and remove the cache.
This is the result of initially looking up the uid in authLdap_user_role and then putting the lookup in authLdap_login into a separate function and reusing it.